finesse/gui/serial_view.py

```diff
@@ -1,138 +1,206 @@
         """
-        layout = QGridLayout()
-        for i, (name, details) in enumerate(self._devices.items()):
-            # Adding the device name
-            layout.addWidget(QLabel(name), i, 0)
-
-            # Its port
-            _port = QComboBox()
-            _port.addItems(list(avail_ports))
-            _port.setCurrentText(details["port"])
-            _port.currentTextChanged.connect(  # type: ignore
-                partial(self.on_port_changed, device_name=name)
-            )
-            layout.addWidget(_port, i, 1)
-
-            # Its baud rate
-            _brate = QComboBox()
-            _brate.addItems([str(br) for br in avail_baud_rates])
-            _brate.setCurrentText(details["baud_rate"])
-            _brate.currentTextChanged.connect(  # type: ignore
-                partial(self.on_baud_rate_changed, device_name=name)
-            )
-            layout.addWidget(_brate, i, 2)
-
-            # And a manual way to open/close it
-            _open_close_btn = QPushButton("Open")
-            _open_close_btn.setCheckable(True)
-            _open_close_btn.released.connect(  # type: ignore
-                partial(
-                    self.on_open_close_clicked, device_name=name, button=_open_close_btn
-                )
-            )
-            _open_close_btn.setChecked(False)
-            layout.addWidget(_open_close_btn, i, 3)
+        self.name = device.name
+        self.label = device.label
 
-        return layout
+        # Add a label showing the device name
+        layout.addWidget(QLabel(self.label), row, 0)
 
-    def on_port_changed(self, new_port: str, device_name: str) -> None:
-        """Callback to deal with a change of port for the given device.
+        self.ports = QComboBox()
+        """The available serial ports for the device."""
+        self.ports.addItems(avail_ports)
+        layout.addWidget(self.ports, row, 1)
 
-        Args:
-            new_port: The new port selected.
-            device_name: Name of the device affected.
-        """
-        logging.info(
-            f"{device_name} - old port: {self._devices[device_name]['port']} "
-            f"- new port: {new_port}"
-        )
-        self._devices[device_name]["port"] = new_port
-
-    def on_baud_rate_changed(self, new_baud_rate: str, device_name: str) -> None:
-        """Callback to deal with a change of baud rate for the given device.
-
-        Args:
-            new_baud_rate: The new port selected.
-            device_name: Name of the device affected.
-        """
-        logging.info(
-            f"{device_name} - old baud_rate: {self._devices[device_name]['baud_rate']} "
-            f"- new baud_rate: {new_baud_rate}"
-        )
-        self._devices[device_name]["port"] = new_baud_rate
+        # Try to load the port from settings
+        if avail_ports:
+            saved_port = cast(
+                str, settings.value(f"{_KEY_PREFIX}/{self.name}/port", avail_ports[0])
+            )
```

Comment:
This an odd construction. Why are you using `_KEY_PREFIX` is it refers always to `serial`? You are indeed using `serial` directly in the `pubsub` commands.
```suggestion
            saved_port = cast(
                str, settings.value(f"serial/{self.name}/port", avail_ports[0])
            )
```